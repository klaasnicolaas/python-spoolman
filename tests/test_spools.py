"""Test the Spool model for Spoolman."""

from __future__ import annotations

from typing import TYPE_CHECKING

from aresponses import ResponsesMockServer
from syrupy.assertion import SnapshotAssertion

from . import load_fixtures

if TYPE_CHECKING:
    from spoolman import Spoolman


async def test_get_spools(
    aresponses: ResponsesMockServer,
    spoolman_client: Spoolman,
    snapshot: SnapshotAssertion,
) -> None:
    """Test getting a list of spools."""
    aresponses.add(
        "192.168.x.x:7912",
        "/api/v1/spool",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("spools.json"),
        ),
    )
    spools = await spoolman_client.get_spools()
    assert spools == snapshot


async def test_get_spool(
    aresponses: ResponsesMockServer,
    spoolman_client: Spoolman,
    snapshot: SnapshotAssertion,
) -> None:
    """Test getting a specific spool."""
    aresponses.add(
        "192.168.x.x:7912",
        "/api/v1/spool/1",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("single_spool.json"),
        ),
    )
    spool = await spoolman_client.get_spool(spool_id=1)
    assert spool == snapshot
