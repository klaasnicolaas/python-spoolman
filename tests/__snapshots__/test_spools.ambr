# serializer version: 1
# name: test_get_spool
  Spool(id=1, filament=Filament(id=1, name='Bambu Green', color='00AE42', vendor=Vendor(id=1, name='Bambu Lab', external_id='Bambu Lab', registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc)), external_id='bambulab_pla_bambugreen_1000_175_n', registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc), material='PLA', density=1.24, diameter=1.75, weight=1000.0, spool_weight=250.0, extruder_temp=220, bed_temp=60), initial_weight=1000.0, spool_weight=250.0, used_weight=0.0, used_length=0.0, remaining_weight=1000.0, remaining_length=335283.6194167644, archived=False, registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc), first_used=datetime.datetime(2024, 12, 28, 23, 0, tzinfo=datetime.timezone.utc), last_used=datetime.datetime(2024, 12, 28, 23, 39, 49, 300000, tzinfo=datetime.timezone.utc))
# ---
# name: test_get_spools
  list([
    Spool(id=1, filament=Filament(id=1, name='Bambu Green', color='00AE42', vendor=Vendor(id=1, name='Bambu Lab', external_id='Bambu Lab', registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc)), external_id='bambulab_pla_bambugreen_1000_175_n', registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc), material='PLA', density=1.24, diameter=1.75, weight=1000.0, spool_weight=250.0, extruder_temp=220, bed_temp=60), initial_weight=1000.0, spool_weight=250.0, used_weight=0.0, used_length=0.0, remaining_weight=1000.0, remaining_length=335283.6194167644, archived=False, registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc), first_used=datetime.datetime(2024, 12, 28, 23, 0, tzinfo=datetime.timezone.utc), last_used=datetime.datetime(2024, 12, 28, 23, 39, 49, 300000, tzinfo=datetime.timezone.utc)),
    Spool(id=2, filament=Filament(id=2, name='PolyTerraâ„¢ Forest Green', color='519f61', vendor=Vendor(id=2, name='Polymaker', external_id='Polymaker', registered=datetime.datetime(2024, 12, 28, 23, 32, 22, tzinfo=datetime.timezone.utc)), external_id='polymaker_pla_polyterraforestgreen_1000_175_n', registered=datetime.datetime(2024, 12, 28, 23, 32, 22, tzinfo=datetime.timezone.utc), material='PLA', density=1.24, diameter=1.75, weight=1000.0, spool_weight=140.0, extruder_temp=210, bed_temp=50), initial_weight=1000.0, spool_weight=140.0, used_weight=0.0, used_length=0.0, remaining_weight=1000.0, remaining_length=335283.6194167644, archived=False, registered=datetime.datetime(2024, 12, 28, 23, 32, 22, tzinfo=datetime.timezone.utc), first_used=None, last_used=None),
  ])
# ---
